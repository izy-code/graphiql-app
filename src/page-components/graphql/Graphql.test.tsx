import { render } from '@testing-library/react';

import GraphQlPage from '@/app/[locale]/GRAPHQL/[[...slug]]/page';

vi.mock('next/navigation', async (importOriginal) => {
  const actual = await importOriginal<object>();

  return {
    ...actual,
    useRouter: vi.fn(() => ({ push: vi.fn((arg: string): string => arg) })),
  };
});

vi.mock('@/hooks/useAuth', () => ({ useAuth: vi.fn((): boolean => true) }));

describe.skip('GraphQL page', () => {
  it('renders correctly', () => {
    const { container } = render(<GraphQlPage />);

    expect(container).toMatchSnapshot();
  });
});
