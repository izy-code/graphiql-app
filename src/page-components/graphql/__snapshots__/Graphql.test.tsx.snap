// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GraphQL page > renders correctly and allows user to enter an endpoint and query to make a request and get a response 1`] = `
<div>
  <div
    class="_page_d43d97"
  >
    <div
      class="_container_d43d97"
    >
      <h1
        class="_title_d43d97"
      >
        GraphQL client
      </h1>
      <div
        class="_items_1663f1"
      >
        <h2
          class="_sectionTitle_1663f1"
        >
          URL
        </h2>
        <div
          class="MuiFormControl-root MuiTextField-root _input_71d148 css-136h9lv-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            Endpoint URL
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              type="text"
              value="https://rickandmortyapi.com/graphql"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Endpoint URL
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root _input_71d148 css-136h9lv-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r1:"
            id=":r1:-label"
          >
            Schema URL
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r1:"
              type="text"
              value="https://rickandmortyapi.com/graphql?sdl"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Schema URL
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="_buttonsContainer_1663f1"
        >
          <button
            class="_button_1c0f93 _button--primary_1c0f93"
            type="button"
          >
            Request
          </button>
          <button
            class="_button_1c0f93 _button--secondary_1c0f93"
            type="button"
          >
            Schema request
          </button>
        </div>
      </div>
      <div
        class="_section_9d89cd"
      >
        <div
          class="_item_9d89cd"
        >
          <div
            class="_wrapper_a5b21b"
          >
            <h4
              class="_title_a5b21b"
            >
              Query: 
            </h4>
            <div
              class="_controls_a5b21b"
            >
              <button
                class="_button_1c0f93 _button--tertiary_1c0f93 _prettifyBtn_a5b21b"
                type="button"
              >
                <span
                  class="visually-hidden"
                >
                  Prettify
                </span>
              </button>
            </div>
          </div>
          <input
            data-testid="codemirror"
            value="query RickCount{  characters(filter: { name: "rick" }) {    info {      count    }  }}"
          />
        </div>
        <h2
          class="_sectionTitle_9d89cd"
        >
          Params
        </h2>
        <div
          class="_container_8cb11c"
        >
          <div
            class="_titleContainer_8cb11c"
          >
            <h4
              class="_subtitle_8cb11c"
            >
              Headers
              :
            </h4>
            <button
              class="_button_1c0f93 _button--primary_1c0f93 _hideBtn_8cb11c _hidden_8cb11c"
              type="button"
            >
              <span
                class="visually-hidden"
              />
            </button>
          </div>
        </div>
        <div
          class="_item_9d89cd"
        >
          <div
            class="_wrapper_a5b21b"
          >
            <h4
              class="_title_a5b21b"
            >
              Variables: 
            </h4>
            <div
              class="_controls_a5b21b"
            >
              <button
                class="_button_1c0f93 _button--tertiary_1c0f93 _prettifyBtn_a5b21b"
                type="button"
              >
                <span
                  class="visually-hidden"
                >
                  Prettify
                </span>
              </button>
              <button
                class="_button_1c0f93 _button--primary_1c0f93 _hideBtn_a5b21b _hidden_a5b21b"
                type="button"
              >
                <span
                  class="visually-hidden"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="_section_22b2aa _response_22b2aa"
    >
      <h2
        class="_sectionTitle_22b2aa"
      >
        GraphQL client
      </h2>
      <div
        class="_oneLine_22b2aa"
      >
        <h4
          class="_status_22b2aa"
        />
        200
      </div>
      <div
        class="_body_22b2aa"
      >
        <div
          class="_wrapper_a5b21b"
        >
          <h4
            class="_title_a5b21b"
          />
          <div
            class="_controls_a5b21b"
          />
        </div>
        <input
          data-testid="codemirror"
          value="{
  "characters": {
    "info": {
      "count": 107
    }
  }
}"
        />
      </div>
    </div>
  </div>
  ;
</div>
`;
